### ISSUE TITLE
Invalid $number parameter: "" cannot be interpreted as a number

### ISSUE THREAD

**Issue by zuk3975** [PrestaShop/PrestaShop#22108]
#### Describe the bug
error appears when entering Admin `Sell > Catalog > Products` while in all shop context

#### Expected behavior

no error.

#### Steps to Reproduce

1. ( have multiple shops available) Select "All shops" context in top right corner
2. got to Sell > Catalog > Products.
3. **EDIT** Click on any product (to open edit page)
4. See error (screenshot)

**Screenshots**
![Screenshot from 2020-11-25 17-47-44](https://user-images.githubusercontent.com/31609858/100252077-27fd6200-2f48-11eb-9a17-5026eb3d4e9f.png)

#### Additional information

* PrestaShop version: develop
* PHP version: 7.3

**Comment by zuk3975** [comment]
Same happens when shop group is selected

**Comment by hibatallahAouadni** [comment]
Hello @zuk3975 

I didn't manage to reproduce your issue with develop, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/100257085-d18f2400-2f45-11eb-97e5-34ed1273c00e.png)

Please check and feedback.
And could you please provide more infos to reproduce it.

Thanks!

**Comment by zuk3975** [comment]
My bad I did not write full path.
The error appears when you click on one of products from list.

(btw are you sure you use latest develop branch, because I've just updated it and I see different "status" (screenshot bellow) in list :man_shrugging: )

![Screenshot from 2020-11-25 18-50-40](https://user-images.githubusercontent.com/31609858/100259429-e02f0880-2f50-11eb-8459-c1c65e74fa52.png)

**Comment by Tibase** [comment]
I have the same issue after update to 1.7.7.0

**Comment by zuk3975** [comment]
It seems to be gone on develop branch :man_shrugging:

**Comment by hibatallahAouadni** [comment]
Hello @Tibase 

I didn't manage to reproduce the issue, I have even changed the product quantity, see the attached screen record below:

https://drive.google.com/file/d/1GuoejRfurnm0GCQ1cvjLYP00H3q3Pd4x/view

Please check and feedback.

Thanks!

**Comment by Fixmetal** [comment]
> I have the same issue after update to 1.7.7.0

me too

**Comment by matks** [comment]
From the error it seems to be an issue that is triggered when prestashop tries to displays products in a specific currency, and it fails because it does not know how to display it following the currency rule.

@Fixmetal @Tibase can you tell us what currencies are enabled on your shops?

**Comment by Fixmetal** [comment]
As for me, it's just €. I'm suspecting something went wrong during the 1.7.6.9=>1.7.7.0 upgrade. I have at least 2 other bugs that I can't seem to find here on github. Right now I'm downgrading, but boy we did modifications meanwhile.

**Comment by WartoWiedziec** [comment]
me too, 
so, it's any solution for this bug?
PrestaShop version: 1.7.7.0 runing
PHP version: 7.3

**Comment by Fixmetal** [comment]
As for me, it was due to a missing **complete** upgrade from 1.7.6.9 - I discovered it timed out whilst upgrading the database. I manually updated it by altering the specific seed SQL to match my database name/prefix table names.

On another note, I had problems whilst changing PHP version from 7.2 to 7.3 into Cpanel due to the default extensions/name changing 7.3 brings with it. Please note that some of them are explicitly reported in [System Requirements](https://devdocs.prestashop.com/1.7/basics/installation/system-requirements/) page, but some of them seem to be specific to the PHP version (and one dependency of another).
I had to enable at least the followings:
- bcmath
- dom
- fileinfo
- gd
- intl
- json
- mbstring
- mysqlnd (is a dependency of nd_mysqli)
- nd_mysqli
- nd_pdo_mysql
- pdo (is a dependency of nd_pdo_mysql)

Please note that my CPanel doesn't have explicit `curl`, `iconv`, `simpleXML`, `openssl` extension checkboxes, but a `php -m` confirms they are enabled:
```
# php -m|egrep -i "simplexml|openssl|iconv|curl"
curl
iconv
openssl
SimpleXMLh
```
More extensions I've enabled for which I honestly can't find documentation whether eventually necessary or not:
- geoip - don't know whether if it's being used by GeoIP PS integration
- imagick - don't know whether it's being used by PS image handler
- opcache - don't know whether it's being used by PS cache integrated system
- pdf - don't know whether it's being used by PS "Make PDF receipt" algorithm

TL;DR
This seems not to be a bug but rather a consequence of a missing full upgrade.

Hope this helps out some folks out there.

**Comment by hibatallahAouadni** [comment]
Hello @zuk3975 

Do you still encounter this issue? 
@Fixmetal Thanks for the details provided :wink: 
@matks do you have any idea what is causing this issue?

Thanks!

**Comment by zuk3975** [comment]
It seems to be gone in develop branch, cannot reproduce anymore.

**Comment by hibatallahAouadni** [comment]
Hello @zuk3975 

Glad to hear :tada: 
@WartoWiedziec could you please enable the dependencies mentionned above by @Fixmetal and feed back :wink: 
I'm closing this issue cause it seems not related to PrestaShop Core.
Please feel free to open a new issue if you encounter another problem.

Thanks!

**Comment by WartoWiedziec** [comment]
I solved this problem by a paid order to the PrestaShop developer => https://sensbit.pl 
I have the problem fixed.

**Comment by BoutiquesInternet** [comment]
Hi, 

The problem is still there on a fresh install 1.7.7.0 with multistore enable.
It's appears on all stores context but not on single shop context.
